dask/dataframe/io/parquet/arrow.py

```diff
@@ -1036,3 +801,1075 @@ def write_metadata(parts, fmd, fs, path, append=False, **kwargs):
+                                # Only think about bailing if we don't need
+                                # stats for filtering
+                                if cmin is None or (last and cmin < last):
+                                    # We are collecting statistics for divisions
+                                    # only (no filters) - Column isn't sorted, or
+                                    # we have an all-null partition, so lets bail.
+                                    #
+                                    # Note: This assumes ascending order.
+                                    #
+                                    gather_statistics = False
+                                    file_row_group_stats = {}
+                                    file_row_group_column_stats = {}
+                                    break
+
+                            if single_rg_parts:
+                                s["columns"].append(
+                                    {
+                                        "name": name,
+                                        "min": pd.Timestamp(cmin)
+                                        if isinstance(cmin, datetime)
+                                        else cmin,
+                                        "max": pd.Timestamp(cmax)
+                                        if isinstance(cmax, datetime)
+                                        else cmax,
+                                        "null_count": cnull,
+                                    }
+                                )
+                            else:
+                                cstats += [cmin, cmax, cnull]
+                            cmax_last[name] = cmax
+                        else:
+                            if single_rg_parts:
+                                s["columns"].append({"name": name})
+                            else:
+                                cstats += [None, None, None]
+                    if gather_statistics:
+                        file_row_group_stats[fpath].append(s)
+                        if not single_rg_parts:
+                            file_row_group_column_stats[fpath].append(tuple(cstats))
+
+        # Construct `parts` and `stats`
+        parts = []
+        stats = []
+        if split_row_groups:
+            # Create parts from each file,
+            # limiting the number of row_groups in each piece
+            split_row_groups = int(split_row_groups)
+            for filename, row_groups in file_row_groups.items():
+                row_group_count = len(row_groups)
+                for i in range(0, row_group_count, split_row_groups):
+                    i_end = i + split_row_groups
+                    rg_list = row_groups[i:i_end]
+                    # Get full path (empty strings should be ignored)
+                    full_path = fs.sep.join(
+                        [p for p in [data_path, filename] if p != ""]
+                    )
```

Comment:
Is this part still fully needed with the latest changes?

Reply:
Good question - We do not need the `full_path` or `rg_list` when we pass an IO-ready fragment object in the graph.

However, we are **not** always reading from fragments in `ArrowDatasetEngine`, and will **never** do so in Dask-CuDF (note that the `read_from_path` option is largely motivated by my desire to re-use most of the `ArrowDatasetEngine` funcitonality in Dask-CuDF).

With the above in mind, it is true that we should probably avoid the creation of `full_path` and `rg_list` in cases where we know they will be ignored.